### DISCUSSION TITLE
Specific usage of the  getStatus method?

### DISCUSSION THREAD

**Discussion by skyerGalaxy** [Makuna/DFMiniMp3#161] [Q&A] [answered]
Thank you for developing this library make me easy to control dfplayer. 

I'm new to arduino.I want to use getStatus method  to Conditional Judgment,just like below(I'm using 1.0.7 vision)
![image](https://github.com/Makuna/DFMiniMp3/assets/90626136/4e34bf51-b6d8-42ab-aac1-d7710d9f94c7)

I checked the information, but I don't have the specific usage of 'getStatus` method, can you help me? Thank you!

**Comment by Makuna** [comment, marked as answer]
(moved to discussions where questions are asked and answered)

[The Wiki Entry for the getStatus method](https://github.com/Makuna/DFMiniMp3/wiki/API-Reference#dfmp3_status-getstatus)

Using the form of `mp3.getStatus().state` is not a good idea in any code.  Every time you call getStatus() it communicates to the chip in this case, in other similar uses it could be far worse.  Don't assume these calls are cheap.  Call it once and store the results then use the results.

Why did you think you needed the single quotes around DfMp3_StatusState_Paused?  DfMp3_Status.state is an enum.  Search for how c++ enums are used.

```
DfMp3_Status status = mp3.getStatus();

if (status.state == DfMp3_StatusState_Playing)
{
}
```

**Reply by skyerGalaxy** [reply]
Thank you sir,I will look it
